### PR TITLE
Sniff for bad indentation inside phpdoc comments.

### PR DESCRIPTION
**Pull request by maresja1**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Please provide test

**Comment by maresja1** [comment]
There you are.

**Comment by TomasVotruba** [comment]
Thanks. It's difficult to spot the difference in tested files. Could you limit their source to the minimal required code?

**Comment by maresja1** [comment]
The difference is between usage of tabs and spaces to indent comments, that is always hard to spot. But hey, I will add two more tests, that will be clearer. There is never enough of tests ;).

**Comment by TomasVotruba** [comment]
I meant to simplify already existing one. It has no added value here, to cover single use case twice.

It will be ready to merge then.

**Comment by maresja1** [comment]
It certainly has added value to test different transition systems. Are you so sure that these two tests cover the same transition system?

**Comment by TomasVotruba** [comment]
I don't know what the intention. There is only one new method, that check some inside indentation of docblock probably. Thus 1 test case (good indentation, bad indentation).

Or are there more possible ways to make it wrong/correct?

**Comment by maresja1** [comment]
Well, that depends what you use unit tests for. If you use them for testing, then covering multiple different setups is beneficial. State of the system can change after first processed line and the next line can be processed differently. The same applies to regression testing.

**Comment by TomasVotruba** [comment]
Back to the issue, what's the simplest use case of this change?

**Comment by maresja1** [comment]
The simplest use case is in the first test. That test is actually based on what I needed this feature for. I also (as per your request) provided you with an even simpler test. At this point, I'm not sure I can help any more.

**Comment by TomasVotruba** [comment]
Could you past the line here?

File: src/ZenifyCodingStandard/Sniffs/WhiteSpace/OperatorSpacingSniff.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ public function process(PHP_CodeSniffer_File $file, $position)
 		$isSpaceAfter = ($this->tokens[$position + 1]['code'] === T_WHITESPACE);
 		$isNewlineAfter = ($this->tokens[$position]['line'] !== $this->tokens[$position + 2]['line']);
 
-		if ( ! $isSpaceBefore || ! $isSpaceAfter || $isNewlineAfter) {
+		if ( ! $isSpaceBefore || ! ( $isSpaceAfter || $isNewlineAfter ) ) {
```
Comment (TomasVotruba) [inline src/ZenifyCodingStandard/Sniffs/WhiteSpace/OperatorSpacingSniff.php]: There is no sniff for this yet, but there are only spaces around `!`, not inside `()`.

This is correct then:
```php
`if ( ! $isSpaceBefore || ! ($isSpaceAfter || $isNewlineAfter)) {
```

File: tests/Sniffs/WhiteSpace/DocBlock/DocBlockSniffTest.php [hunk 2]
```diff
@@ -20,4 +20,22 @@ public function testDetection()
...
+		$this->assertSame(2, $codeSnifferRunner->getErrorCountInFile(__DIR__ . '/wrong-inside.php'));
+		$this->assertSame(0, $codeSnifferRunner->getErrorCountInFile(__DIR__ . '/correct-inside.php'));
+	}
+
+
+	public function testIdentationInside2()
+	{
+		$codeSnifferRunner = new CodeSnifferRunner('ZenifyCodingStandard.WhiteSpace.DocBlock');
+
+		$this->assertSame(1, $codeSnifferRunner->getErrorCountInFile(__DIR__ . '/wrong-inside2.php'));
```
Comment (TomasVotruba) [inline tests/Sniffs/WhiteSpace/DocBlock/DocBlockSniffTest.php]: These can be added to previous method. As it's standard, that tests are per sniff.

File: tests/Sniffs/WhiteSpace/DocBlock/correct-inside.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * @ORM\Entity
+ * @ORM\Table(name="test")
+ * @ORM\InheritanceType("JOINED")
+ * @ORM\DiscriminatorColumn(name="discriminator", type="text")
+ * @ORM\DiscriminatorMap({
+ * 		"complex" = "Test1",
+ * 		"simple" = "Test2"
+ * })
+ */
+class Test
```
Comment (TomasVotruba) [inline tests/Sniffs/WhiteSpace/DocBlock/correct-inside.php]: This test file can be removed, as it's not readable and the feature is covered by `correct-inside2.php` already.

File: tests/Sniffs/WhiteSpace/DocBlock/correct-inside2.php [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+/**
+ * Class Test
+ * 	indented text (one space followed by tabs)
```
Comment (TomasVotruba) [inline tests/Sniffs/WhiteSpace/DocBlock/correct-inside2.php]: Very nice!

File: tests/Sniffs/WhiteSpace/DocBlock/wrong-inside.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ * @ORM\Entity
+ * @ORM\Table(name="test")
+ * @ORM\InheritanceType("JOINED")
+ * @ORM\DiscriminatorColumn(name="discriminator", type="text")
+ * @ORM\DiscriminatorMap({
+ *                "complex" = "Test1",
+ *                "simple" = "Test2"
+ * })
+ */
+class Test
```
Comment (TomasVotruba) [inline tests/Sniffs/WhiteSpace/DocBlock/wrong-inside.php]: This can be remove too.

File: tests/Sniffs/WhiteSpace/DocBlock/wrong-inside2.php [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+/**
+ * Class Test
+ *     indented text (all spaces)
```
Comment (TomasVotruba) [inline tests/Sniffs/WhiteSpace/DocBlock/wrong-inside2.php]: Great!

File: tests/Sniffs/WhiteSpace/OperatorSpacing/wrong.php [hunk 7]
```diff
@@ -2,5 +2,5 @@
 
 $result = 5 &&3|| 2;
 
-$car = 'wheels' +
+$car = 'wheels' .
```
Comment (TomasVotruba) [inline tests/Sniffs/WhiteSpace/OperatorSpacing/wrong.php]: Very fine! Thanks

**Comment by TomasVotruba** [comment]
I wrote final notes. There are only 3 small changes and it's ready to merge!

**Comment by maresja1** [comment]
Should be resolved.

**Comment by TomasVotruba** [comment]
Thank you!

**Comment by maresja1** [comment]
Thank you!

**Pull request merged**